ludwig/features/numerical_feature.py

```diff
@@ -60,6 +61,8 @@ def get_feature_meta(column, preprocessing_parameters):
                     'min': column.astype(np.float32).min(),
                     'max': column.astype(np.float32).max()
                 }
+            elif preprocessing_parameters['normalization'] == 'log1p':
+                return {}
```

Comment:
We could avoid this if by adding an additional function to the transformer interface like `fit_transform_params` like:
```
def fit_transform_params(column):
   ...
   return {calculated_values_like_min_and_max}
```

Reply:
I understand the comment is to add a `fit_transform_params()` method to the numeric transformer classes.

I'd like to discuss comment.  I have question and concern.

**Question**: If I understand the intent, we replace this code in the `get_feature_metadata()`
```
        if preprocessing_parameters['normalization'] is not None:
            if preprocessing_parameters['normalization'] == 'zscore':
                return {
                    'mean': column.astype(np.float32).mean(),
                    'std': column.astype(np.float32).std()
                }
            elif preprocessing_parameters['normalization'] == 'minmax':
                return {
                    'min': column.astype(np.float32).min(),
                    'max': column.astype(np.float32).max()
                }
            elif preprocessing_parameters['normalization'] == 'log1p':
                return {}
            else:
                logger.info(
                    'Currently zscore and minmax are the only '
                    'normalization strategies available. No {}'.format(
                        preprocessing_parameters['normalization'])
                )
                return {}
        else:
            return {}
```

Does `get_feature_metadata()` become something like this?

```
def get_feature_metadata(column, preprocessing_parameters):

     # some code to instantiate  the numeric transformer object
        numeric_transformer = get_from_registry(
            preprocessing_parameters.get('normalization', None),
            numeric_transformation_registry
        )(**metadata[feature[NAME]])

      return numeric_transformer.fit_transform_params(column)
```

**Concern**:  If the above is directionally correct, then it is not clear to me how `# some code to instantiate the numeric transformer object` will work.  For two of the transformer classes, the constructor needs to know the pre-computed values for `mean`/`std` or `min`/`max`.  This seems like a chicken and egg kind of situation.

One possibility to get around this is to change the constructor `def __init__()` method to call `fit_transform_params()`.  This should work during pre-processing because we have values in the original value space.  However, I don't see how this work in post-processing because we only have values in the transformed space and we do not save the  numeric transformer object.  The numeric transformer object is instantiated in both pre-processing and post-processing.

Anyway, I think there are some nuances that need to be worked out with the `fit_transform_params()` method.

Reply:
I see your point. i believe it is solved by making the fit function a static/class function that doesn't require the class to be instantiated to be used.

Reply:
I have a proposed solution for the `fit_transform_params()` method.  However to get it to work, even with a static `fit_transform_params()`, I had to handle two different situations when the the numeric transformer object is instantiated.  Pay close attention to comments in the `__init__()` and parameter validations performed in `transform()` and `Inverse_transform()` methods.

Does this look reasonable?

Reply:
I'm guessing you are refferring to `# When parameters are None we only need object to use
         # the fit_transform_params method, other methods should not be used`
I believe we don't need this additional consideration, swe don't need the parameters to be set to None by default, and also we don't need to check if they are not none. i think the doubt stems from line 56 where you initialize an obect before calling the static method, but you actually do not need to initialize it, you can call the `fit_transform_params` directly from the call reference you get from the registry. This should solve the outstanding doubts i believe.

Reply:
learning something new every day.  done.